aiobungie/internal/factory.py

```diff
@@ -124,14 +124,14 @@ def deserialize_destiny_membership(
             id=int(payload["membershipId"]),
             name=name,
             code=payload.get("bungieGlobalDisplayNameCode", None),
-            last_seen_name=payload.get("LastSeenDisplayName", payload["displayName"]),
+            last_seen_name=payload.get("LastSeenDisplayName", payload.get("displayName", name)),
             type=enums.MembershipType(payload["membershipType"]),
             is_public=payload["isPublic"],
             crossave_override=enums.MembershipType(payload["crossSaveOverride"]),
             icon=assets.Image(payload.get("iconPath", "")),
             types=[
                 enums.MembershipType(type_)
-                for type_ in payload["applicableMembershipTypes"]
+                for type_ in payload.get("applicableMembershipTypes", {0:0})
             ],
```

Comment:
This is redundant and useless object creation, You should check if the `applicableMembershipTypes` is in the payload otherwise leave it as empty list.

```suggestion
            types=[
                enums.MembershipType(type_)
                for type_ in payload["applicableMembershipTypes"]
                if "applicableMembershipTypes" in payload
            ],
```